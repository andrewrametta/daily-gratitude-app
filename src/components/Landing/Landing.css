.quote {
  font-size: 16px;
  margin-left: 10%;
  margin-right: 10%;
  margin-bottom: 30px;
  color: rgb(84, 84, 84);
}

.landing-container {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: space-evenly;
}

.landing-item {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex: 1;
}

img .landing-img {
  max-width: 200px;
  max-height: 200px;
}

section .landing-section {
  flex: 1;
  display: flex;
  flex-direction: column;
  max-width: 200px;
  max-height: 200px;
}

.section-child {
  flex: 1;
  margin-bottom: 20px;
  padding: 15px;
  border-radius: 15px;
  background-color: white;
  margin-left: 15px;
  margin-right: 15px;
  border: none;
}

.button-container {
  display: flex;
  flex-direction: column;
  text-align: center;
  margin-bottom: 30px;
  margin-top: 20px;
}

.button-register {
  margin-bottom: 30px;
  padding: 5px;
  border-radius: 15px;
  border: none;
  background-color: #3dc4c0;
  color: white;
}
.button-demo {
  margin-bottom: 10px;
  padding-top: 5px;
  padding-bottom: 5px;
  padding-left: 10px;
  padding-right: 10px;
  border-radius: 15px;
  border: none;
  background-color: #3dc4c0;
  color: white;
}

p {
  color: rgb(138, 138, 138);
}

@media all and (min-width: 700px) {
  .landing-container {
    flex-direction: row;
  }
  section .landing-section {
    flex: 1;
  }

  .landing-img-container {
    display: flex;
    justify-content: space-evenly;
  }

  img .landing-img {
    width: 200px;
    height: 200px;
  }
}
