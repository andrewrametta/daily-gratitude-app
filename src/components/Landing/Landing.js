import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./Landing.css";

class Landing extends Component {
  render() {
    return (
      <>
        <p className="quote">
          "Something as simple as writing down three things you are grateful for
          every day for 21 days in a row significantly increases your level of
          optimism, and it holds for the next six months." -- Shawn Achor
        </p>
        <div className="landing-container">
          <div className="landing-item">
            <img
              className="landing-img"
              src="/mountain.jpg"
              alt="Daily Gratitude"
            />
            <section className="landing-section">
              <h3 className="card">Change your outlook</h3>
              <p>
                We are in some of the most difficult times in modern history.
                People are suffering from anxiety and depression at higher
                levels than ever documented before.
              </p>
            </section>
          </div>
          <div className="landing-item">
            <img
              className="landing-img"
              src="/gothis.jpg"
              alt="Daily Gratitude"
            />
            <section className="landing-section">
              <h3 className="card">How does it work?</h3>

              <p>
                You can sign up and create a profile and then start your own
                gratitude journal. Each day you can create a new entry. There is
                also a dashboard page to view your entire journal.
              </p>
            </section>
          </div>
          <div className="landing-item">
            <img
              className="landing-img"
              src="/yoga.jpg"
              alt="Daily Gratitude"
            />
            <section className="landing-section">
              <h3 className="card">Want to test it out?</h3>

              <p>
                We have set up a demo account so that you can test the app.
                Click the Demo link below and you will land on the login page.
                Everything will be auto filled for you, just click login and
                enjoy!
              </p>
            </section>
          </div>
        </div>
        <div className="button-container">
          <Link to="/login">
            <button className="button-demo">Demo</button>
          </Link>

          <Link to="/register">
            <button className="button-register">Register</button>
          </Link>
        </div>
      </>
    );
  }
}

export default Landing;
