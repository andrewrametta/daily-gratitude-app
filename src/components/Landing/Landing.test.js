import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Landing from "./Landing";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe(`Landing component`, () => {
  it("renders the complete page", () => {
    const wrapper = shallow(<Landing />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
