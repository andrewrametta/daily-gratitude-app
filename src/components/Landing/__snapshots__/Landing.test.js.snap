// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders the complete page 1`] = `
<Fragment>
  <p
    className="quote"
  >
    "Something as simple as writing down three things your are grateful for every day for 21 days in a row significantly increases your level of optimism, and it holds for the next six months." -- Shawn Achor
  </p>
  <div
    className="landing-container"
  >
    <div
      className="landing-item"
    >
      <img
        alt="Daily Gratitude"
        className="landing-img"
        src="/mountain.jpg"
      />
      <section
        className="landing-section"
      >
        <h3
          className="card"
        >
          Change your outlook
        </h3>
        <p>
          We are in some of the most difficult times in modern history. People are suffering from anxiety and depression at higher levels than ever documented before.
        </p>
      </section>
    </div>
    <div
      className="landing-item"
    >
      <img
        alt="Daily Gratitude"
        className="landing-img"
        src="/gothis.jpg"
      />
      <section
        className="landing-section"
      >
        <h3
          className="card"
        >
          How does it work?
        </h3>
        <p>
          You can sign up and create a profile and then start a your own gratitude journal. Each day you can create a new entry. There is also a dashboard page to view your entire journal.
        </p>
      </section>
    </div>
    <div
      className="landing-item"
    >
      <img
        alt="Daily Gratitude"
        className="landing-img"
        src="/yoga.jpg"
      />
      <section
        className="landing-section"
      >
        <h3
          className="card"
        >
          Want to test it out?
        </h3>
        <p>
          We have set up a demo account so that you can test the app. Click the Demo link below and you will land on the login page. The username and password will be auto filled for you, just click login and enjoy!
        </p>
      </section>
    </div>
  </div>
  <div
    className="button-container"
  >
    <Link
      to="/login"
    >
      <button
        className="button-demo"
      >
        Demo
      </button>
    </Link>
    <Link
      to="/register"
    >
      <button
        className="button-register"
      >
        Register
      </button>
    </Link>
  </div>
</Fragment>
`;
